### DISCUSSION TITLE
Collaboration Guidelines

### DISCUSSION THREAD

**Discussion by nanafox** [codemindshub/liteshell#8] [Announcements]
Hello Collabs,

Here are a few guidelines to help us during the implementation of our `liteshell` project.

## Branches

It is important to create branches when working on this project, whether it's for creating a feature or fixing a bug. Here are the accepted patterns:

### Bug fixes

Undeniably, there will be some bugs that would escape us while building this. If you find a bug, you can raise an issue so the team implementing the feature will work on it. You could also create a Pull Request with the bug fixed. Here's how to name the branch `bugfixes/<bug_name>`. For example, if you are fixing a memory leak error, you could use `bugfixes/memory_leaks`.

### Features

Each team will be assigned features they will have to build. Ensure you are naming your branches to reflect the feature you just created. Here's how to do it: `feature/<feature_name>`

### Updates

Found an interesting or more efficient way to handle an existing feature?  Use thing pattern when creating your branch `updates/<updates_performed>`

## Commit messages

It's important to limit your commit messages to 50 characters maximum. Use the title to state WHAT you did, not HOW or WHY. You can provide additional explanations in the body of the commit message, but try to keep each line within 70 characters. This is where you can explain why you did what you did and how it will benefit the project.

## Pull Requests

It is important to only create Pull Requests after you and your team have finalized the code logic and conducted thorough testing on your own. If untested files are included in the Pull Request, it will take much longer time to review and merge. We do not have automated testing at this point, which is why it is crucial to help make things easier for everyone involved.
When submitting a Pull Request, it's important to explain in detail the feature being suggested and the reasoning behind it.

**Comment by OmondiHillaryA1997** [comment]
Thanks for the Info Mentor.
